app/src/main/java/io/github/jbellis/brokk/AnalyzerWrapper.java

```diff
@@ -59,17 +54,22 @@ public AnalyzerWrapper(IProject project, ContextManager.TaskRunner runner, @Null
         gitRepoRoot = project.hasGit() ? project.getRepo().getGitTopLevel() : null;
         this.runner = runner;
         this.listener = listener;
+        this.watchService = new ProjectWatchService(root, gitRepoRoot, this);
 
         // build the initial Analyzer
         future = runner.submit("Initializing code intelligence", () -> {
+            // Watcher will wait for the future to complete before processing events,
+            // but it will start watching files immediately in order not to miss any changes in the meantime.
+            var delayNotificationsUntilCompleted = new CompletableFuture<Void>();
+            watchService.start(delayNotificationsUntilCompleted);
```

Comment:
This should fix the [race condition](https://github.com/BrokkAi/brokk/issues/501#issuecomment-3096761971) described in the issue.